views/application.py

```diff
@@ -38,3 +43,51 @@ def get_queryset(self):
         """
 
         return Application.objects.filter(team_members=self.request.person)
+
+    def update(self, request, *args, **kwargs):
+        """
+        update membership status of the application member
+        :param request: the request being processed
+        :param args: arguments
+        :param kwargs: keyword arguments
+        :return: updated instance
+        """
+
+        application = kwargs['pk']
+        new_team_members = request.data['team_members']
+        try:
+            application = Application.objects.get(pk=application)
+        except Application.DoesNotExist:
+            pass
+        old_team_members = application.team_members.all()
+        old_team_member_ids = []
+        new_team_member_ids = []
+        for old_member in old_team_members:
+            old_team_member_ids.append(old_member.id)
+        for new_member in new_team_members:
+            new_team_member_ids.append(new_member['id'])
+        partial = kwargs.pop('partial', False)
+        instance = self.get_object()
+        serializer = self.get_serializer(
+            instance,
+            data=request.data,
+            partial=partial
+        )
+        serializer.is_valid(raise_exception=True)
+        self.perform_update(serializer)
+        if getattr(instance, '_prefetched_objects_cache', None):
+            instance._prefetched_objects_cache = {}
+
+        if len(old_team_member_ids) > len(new_team_member_ids):
+            send_membership_notification(
+                application.name,
+                'remove',
```

Comment:
The better thing would be to just send `'removed from'` and `'added to'` in the function call, so that you can directly insert this into the notification template without having to handle the `if-else` condition in the function!